_server/src/CMakeLists.txt

```diff
@@ -5,10 +5,23 @@ add_executable(${SERVER_NAME} ${ALL_SERVER_SOURCES})
 target_include_directories(${SERVER_NAME} PUBLIC ${SERVER_SOURCES_DIR})
 
 # Link libs
+if(TARGET Boost::boost)     #Check is Boost installed handly
```

Comment:
Remove this. It's not needed if boost was integrated to cmake presets using conan

Reply:
@Field-Effect-Transistor at least consider if it's still needed

Reply:
When I remove this peace of code, next error occurs:
```
PS C:\Users\UsVeR\Documents\GitHub\SageStore\build>    cmake .. --preset conan-debug -DBUILD_CLIENT=ON -DBUILD_SERVER=ON
Preset CMake variables:

  CMAKE_BUILD_TYPE="Debug"
  CMAKE_POLICY_DEFAULT_CMP0091="NEW"
  CMAKE_TOOLCHAIN_FILE:FILEPATH="C:\Users\UsVeR\Documents\GitHub\SageStore\build\conan_toolchain.cmake"

-- Using Conan toolchain: C:/Users/UsVeR/Documents/GitHub/SageStore/build/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions ON
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- Could NOT find WrapVulkanHeaders (missing: Vulkan_INCLUDE_DIR) 
-- Conan: Component target declared 'spdlog::spdlog'
-- Conan: Component target declared 'fmt::fmt'
-- Conan: Target declared 'nlohmann_json::nlohmann_json'
-- Conan: Component target declared 'SQLite::SQLite3'
-- Conan: Component target declared 'spdlog::spdlog'
-- Conan: Component target declared 'fmt::fmt'
-- Configuring done (4.6s)
CMake Error at _server/src/CMakeLists.txt:8 (target_link_libraries):
  Target "SageStoreServer" links to:

    boost::boost

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.2s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

As I understand (as GPT explains) it occures because boost::boost is namespace, but Boost::boost is somthing as path to library in file. Also manually installed Boost requires Boost::boost in target_link_libraries(), but conan installed Boost requires boost::boost.

Reply:
So add_library() -- is a function that gives alternative name for some target?
`add_library(boost_common ALIAS Boost::boost)`
And also we should make our own error instead of CMake error?
```
else()
    # Error or fallback handling if neither target exists
    message(FATAL_ERROR "Boost not found")
endif()
```
Correct?

Reply:
@Field-Effect-Transistor no, it add_library to cmake configuration with our custom name. But in our case... It is used to rename it actually...